oasislmf/cmd/api.py

```diff
@@ -0,0 +1,362 @@
+
+import getpass
+import io
+import json
+import sys
+
+from argparse import RawDescriptionHelpFormatter
+from pathlib2 import Path
+
+from ..utils.exceptions import OasisException
+
+from .cleaners import as_path, PathCleaner
+from .base import OasisBaseCommand, InputValues
+
+from ..api_client.client_manager import APIClient
+
+
+def load_credentials(login_arg, logger=None):
+    """
+    Load credentials from JSON file
+
+    Options:
+        1. '--api-server-login ./APIcredentials.json'
+
+        2. Load credentials from default config file
+        '-C oasislmf.json'
+
+        3. Prompt for username / password
+    """
+    if isinstance(login_arg, (str, unicode)):
+        with io.open(login_arg, encoding='utf-8') as f:
+            return json.load(f)
+
+    elif isinstance(login_arg, (dict)):
+        if set(['password','username']) <= set(login_arg.keys()):
+            return login_arg
+
+    else:
+        logger.info('No Login provided - Fallback to prompt')
+
+    try:
+        api_login = dict()
+        api_login['username'] = raw_input('Username: ')
```

Comment:
You're using `raw_input` which presents a user prompt, but I believe this only works for Python 2. I don't know this will still run on Python 3, where `input` replaces `raw_input`.